datafusion/sqllogictest/test_files/case.slt

```diff
@@ -50,3 +50,19 @@ SELECT CASE WHEN a > 2 THEN b END FROM foo
 NULL
 4
 6
+
+# scalar or scalar (string)
+query T
+SELECT CASE WHEN a > 2 THEN 'even' ELSE 'odd' END FROM foo
```

Comment:
Looks like `NULL` handling in this specialized implementation is not tested, we can add a (NULL, NULL) row into foo table

Reply:
Good point. I have added this.